api/envoy/extensions/filters/network/mtls_failure_response/v3/mtls_failure_response.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.network.mtls_failure_response.v3;
+
+import "envoy/type/v3/token_bucket.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.network.mtls_failure_response.v3";
+option java_outer_classname = "MtlsFailureResponseProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/mtls_failure_response/v3;mtls_failure_responsev3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: mTLS Failure Response]
+// [#extension: envoy.filters.network.mtls_failure_response]
+
+// Configuration for the mTLS Failure Response filter. As opposed to returning a SSL failure to a client after failed
+// client cert validation, this filter allows for customization of the response.
+message MtlsFailureResponse {
+  enum ValidationMode {
+    PRESENTED = 0;
```

Comment:
What's the use-case for allowing this when a cert is presented, but not validated?

Reply:
This is similar to https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-routematch-tlscontextmatchoptions. My idea was that, I think a combination could help users pick and match: If no cert is presented, tarpit the connection. If it is present, use it in combination with upstream objects. For eg. with routing, validate a presented cert for route X, but presentation alone is enough for Y.